### PR TITLE
Correct CPUQuota service limit regex

### PR DESCRIPTION
**Pull request by matt6697**
Corrected CPUQuota service limit regex
Added CPuQuota service limit unit test

### CODE REVIEW

**Comment by ekohl** [comment]
You can also specifically test types. See https://rspec-puppet.com/documentation/type_aliases/ for an example.

File: spec/defines/service_limits_spec.rb [hunk 1]
```diff
@@ -48,6 +49,9 @@
           it { is_expected.to create_file("/etc/systemd/system/#{title}.d/90-limits.conf").with(
             :content => /LimitRTPRIO=50/
           ) }
+	  it { is_expected.to create_file("/etc/systemd/system/#{title}.d/90-limits.conf").with(
```
Comment (raphink) [inline spec/defines/service_limits_spec.rb]: Indentation seems wrong here. Can you fix this please?

File: types/servicelimits.pp [hunk 2]
```diff
@@ -20,7 +20,7 @@
     Optional['CPUAccounting']       => Boolean,
     Optional['CPUShares']           => Integer[2,262144],
     Optional['StartupCPUShares']    => Integer[2,262144],
-    Optional['CPUQuota']            => Pattern['^([1-9][0-9]?$|^100)%$'],
+    Optional['CPUQuota']            => Pattern['^([1-9][0-9]?|100)%$'],
```
Comment (raphink) [inline types/servicelimits.pp]: Good catch

Comment (matt6697) [reply types/servicelimits.pp]: Tks ;-)

**Comment by matt6697** [comment]
Corrected the indentation issue
Maybe the test types tests could be integrated in another PR.(not implemented for any variable of the module at the moment).

**Comment by iridian-ks** [comment]
Hello! I just ran into this bug. Any plans on getting it merged in?

**Pull request merged**